pynetdicom/transport.py

```diff
@@ -276,7 +276,16 @@ def ready(self):
             self.event_queue.put('Evt17')
             return False
 
-        return bool(ready)
+        return (
+            bool(ready)
```

Comment:
this may not be strictly correct per https://docs.python.org/dev/library/ssl.html#notes-on-non-blocking-sockets – an OS-level socket may be ready, but the SSL socket may not actually be ready

Reply:
given that this is meant to immediately read "are there available bytes", it might be better to branch away entirely in the case of an `SSLSocket`; don't use `select.select` at all, and instead _just_ check `pending`, as that's what will tell you if there is readable data on the socekt

Reply:
Good point, we might erroneously treat errors in https://github.com/pydicom/pynetdicom/blob/master/pynetdicom/dul.py#L234-L308 as terminal failures. @scaramallion wdyt?

Reply:
This broke when I tried it out in 5fecdd5

Reply:
@r1b, I'm not sure what you mean by treating errors as terminal failures.

As I understand it, the sockets should be blocking by default so `select` should be suitable. But I'm not an expert either...